Sources/SourceKitLSP/Swift/SwiftTestingScanner.swift

```diff
@@ -95,7 +95,15 @@ struct TestingAttributeData {
           return false
         }
       }.flatMap(\.arguments)
-      .compactMap { $0.expression.as(StringLiteralExprSyntax.self)?.representedLiteralValue }
+      .compactMap {
+          if let stringLiteral = $0.expression.as(StringLiteralExprSyntax.self) {
+              return stringLiteral.representedLiteralValue
+          } else if let memberAccess = $0.expression.as(MemberAccessExprSyntax.self) {
+              let baseName = (memberAccess.baseName.map { "\($0)." } ?? "").replacing(#/Tag\./#, with: "")
+              return "\(baseName)\(memberAccess.declName.baseName.text)"
+          }
+          return nil
```

Comment:
I don’t think this handles `Testing.Tag.blue` correctly (or does it? – I think a test case for that would be good). Also, a test case for `Tag.blue` would be good.

What do you think about 
- Introducing a `components: [String]` property on `MemberAccessExprSyntax` like we do for `TypeSyntax` here (using essentially the implementation of `fullyQualifiedName` and `baseName`)
https://github.com/apple/sourcekit-lsp/blob/b9af0cf2ef51bf14bdf1476cbdfd72ea52725f2c/Sources/SourceKitLSP/Swift/SwiftTestingScanner.swift#L365-L377
- Making `fullyQuaifiedName` use the new `components` property and joining the components together using `.`
- And then here you could check something like
```swift
if components.starts(with: ["Testing", "Tag") {
  components = components.dropFirst(2)
} else if components.starts(with: ["Tag") {
  components = components.dropFirst(1)
}
return components.joined(separator: ".")
```

Reply:
Thanks for the suggestions, I certainly like it better than what I had before. I've added a test to cover tags defined as `.foo`, `Tag.foo` and `Testing.Tag.foo`. Also tests Tags defined nested in structs on Tag.